cas-server-core-logout/src/main/java/org/jasig/cas/logout/LogoutManagerImpl.java

```diff
@@ -68,6 +70,16 @@ public LogoutManagerImpl(final LogoutMessageCreator logoutMessageBuilder) {
             LOGGER.info("Single logout callbacks are disabled");
             return logoutRequests;
         }
+        performLogoutForTicket(ticket, logoutRequests);
+
+        final Collection<ProxyGrantingTicket> proxyGrantingTickets = ticket.getProxyGrantingTickets();
```

Comment:
I'd prefer to have `performLogoutForTicket` perform this loop and recur on any PGTs found since this is more a property of the ticket itself. Given how empty this class is, I'd take a long look at moving `performLogout` to the ticket implementation classes. I'm not very familiar with the logout process so I'm not sure how possible that really is.